broadinstitute/single_cell_portal_core
# Create panel manager for explore page (SCP-5215)

Part one for the Facet Filter component - just a refactor of existing code no new functionality and very little new code. I wanted to get this out so any other work being done in these files can work within this refactor and also because this is already a large PR so wanted to split the actual new feature into its' own PR.


This PR restructures the code a little bit to make the addition of the Filter Facet component easier.

The Explore tab consists of (among other things) the Plot (left side) the Panel (right side) and tabs (Scatter, Dot Plot, etc.) The code in `ExploreDisplayTabs` had managed which tabs, plots and panels to show and was almost 900 lines long. Now that we want to add a third right-side Panel option (Facet Filter) it would make the logic more complicated and make this file even longer. So instead I have separated out a bunch of the logic for managing the right-side Panel into a file called `ExploreDisplayPanelManager`. This will then figure out which of the soon to be three panels (Options/Regular, DE, Facet Filter) to display. 


To test this PR:
- There is no new functionality and all the looks should be the same - so pull down this branch and start up a local server.
- Navigate to a study (one with DE will have more cases to test)
- Then compare to Staging or Prod, clicking around, clicking the DE button, changing sizes etc. 
- There should not be a difference between local and Staging or Prod.


Upcoming work - the new Facet Filter panel will be added and the `ExploreDisplayPanelManager` will sort out when to show this.

Notes - I haven't added new tests yet, will plan to incorporate in future PR.

Showing that staging and local are the same looking:
https://github.com/broadinstitute/single_cell_portal_core/assets/54322292/e4e90e92-db8f-49c2-9106-9b13c492b618


![Screenshot 2023-08-28 at 12 26 42 PM](https://github.com/broadinstitute/single_cell_portal_core/assets/54322292/7b9ea53b-b7d0-416e-891b-68ddc5374d94)
